addons/web/static/src/views/calendar/calendar_model.js

```diff
@@ -504,10 +504,11 @@ export class CalendarModel extends Model {
             (fieldMapping.all_day && rawRecord[fieldMapping.all_day]) ||
             false;
 
-        if (isAllDay) {
+        if (rawRecord["allday"] || false) {
             start = start.startOf("day");
             end = end.startOf("day");
         }
+
```

Comment:
Do not use fields in `rawRecord` without the mapping in the main model.
Here, it will work for your case because your field is named `allday` but the field used for computing a "all day" event is not always named `allday`.

Maybe we don't need these lines anymore since we converted calendar to owl.

Reply:
But the problem is that we don't want to set the start and end to midnight just because it's in the allday cell, we (would) want to if they are marked as allday. 
That being said, i have no clue why we would want to set them to midnight anyway so I'll gladly remove them if you agree.

Reply:
Yep, I agree